core/src/main/java/com/github/games647/fastlogin/core/shared/FloodgateManagement.java

```diff
@@ -119,15 +153,12 @@ public void run() {
             }
         }
 
-        if (!isRegistered && !isAutoAuthAllowed(autoRegisterFloodgate)) {
+        if ((!isRegistered && !isAutoAuthAllowed(autoRegisterFloodgate))
+                || (profile.getFloodgate() == FloodgateState.LINKED && !isLinked)
```

Comment:
Could you please explain the different cases and why they are made. For me it's hard to understand what is actually happening here. Maybe you can extract them to readable methods if that helps.